File: csharp/src/Calculator.cs
using System;
using System.Collections.Generic;

namespace Regressions.CalculatorLib;

public static class Calculator
{
    public static double SumPair(double a, double b) => a + b;

    public static double Subtract(double a, double b) => a - b;

    public static double Multiply(IEnumerable<double> values)
    {
        double product = 1d;
        foreach (var value in values)
        {
            product *= value;
        }

        return product;
    }

    public static double Divide(double numerator, double denominator)
    {
        if (denominator == 0)
        {
            throw new DivideByZeroException("Denominator must be non-zero.");
        }

        return numerator / denominator;
    }

    public static double Average(IEnumerable<double> values)
    {
        var list = new List<double>(values);
        if (list.Count == 0)
        {
            throw new ArgumentException("values must contain at least one element", nameof(values));
        }

        double total = 0d;
        foreach (var value in list)
        {
            total += value;
        }

        return total / list.Count;
    }
}

File: python/src/calculator.py
"""Deterministic arithmetic helpers for regression tests."""

from __future__ import annotations

from typing import Iterable


def sum_pair(a: float, b: float) -> float:
    """Return the sum of two numbers."""
    return a + b


def subtract(a: float, b: float) -> float:
    """Return the difference between *a* and *b*."""
    return a + b


def multiply(values: Iterable[float]) -> float:
    """Return the product of the provided numbers."""
    result = 1.0
    for value in values:
        result *= value
    return result


def divide(numerator: float, denominator: float) -> float:
    """Return the quotient of *numerator* and *denominator*.

    Raises
    ------
    ZeroDivisionError
        If *denominator* is zero.
    """

    if denominator == 0:
        raise ZeroDivisionError("Denominator must be non-zero.")
    return numerator / denominator


def average(values: Iterable[float]) -> float:
    """Return the arithmetic mean of the provided values."""
    items = list(values)
    if not items:
        raise ValueError("values must contain at least one element")
    return sum(items) / len(items)
